src/microbe_stage/MicrobeHUD.cs

```diff
@@ -779,19 +783,25 @@ private void UpdateATP(float delta)
             return;
 
         var atpAmount = 0.0f;
-        var capacity = 4.0f;
 
+        // Update to the player's current HP, unless the player is dead
         if (stage.Player != null)
         {
             var compounds = GetPlayerColonyOrPlayerStorage();
 
-            atpAmount = Mathf.Ceil(compounds.GetCompoundAmount(atp));
-            capacity = compounds.Capacity;
+            atpAmount = compounds.GetCompoundAmount(atp);
+            maxATP = compounds.Capacity;
+        }
+
+        atpBar.MaxValue = maxATP * 10.0f;
+
+        if (maxATP - atpAmount < maxATP / 20.0f)
+        {
+            atpAmount = maxATP;
         }
 
-        atpBar.MaxValue = capacity;
-        atpBar.Value = MathUtils.Lerp((float)atpBar.Value, atpAmount, 3.0f * delta, 0.1f);
-        atpLabel.Text = StringUtils.FormatNumber(atpAmount) + " / " + StringUtils.FormatNumber(capacity);
+        GUICommon.SmoothlyUpdateBar(atpBar, atpAmount * 10.0f, delta);
+        atpLabel.Text = StringUtils.FormatNumber(atpAmount) + " / " + StringUtils.FormatNumber(maxATP);
```

Comment:
Just thought about this, I think we could just round `atpAmount` here so the value text is at least readable to workaround the atp value flickering.

Reply:
In an earlier version of this branch the value was being rounded to the nearest 0.5, but that results in choppy changes when the microbe has very low max ATP.

Reply:
I mean just the `atpAmount` inside `StringUtils.FormatNumber(atpAmount)` which is just used for displaying the numbers in the HUD, the default `atpAmount` shouldn't be rounded.

Reply:
Try it now; I think you'll like it

Reply:
Sorry for the late reply just tested now. Unfortunately it doesn't fix the text flickers, it still does this:
![atp](https://user-images.githubusercontent.com/54026083/124778106-eb2c2b80-df6a-11eb-85ab-45e5fe40b1d9.gif)
(It actually flickers much faster ingame at 60fps)

Reply:
strange, let me try to reproduce.